qiskit/transpiler/passes/scheduling/calibration_creators.py

```diff
@@ -254,3 +273,114 @@ def get_calibration(self, params: List, qubits: List) -> Schedule:
+            a Gaussian square pulse on the DriveChannel.
+        """
+        if isinstance(inst[1], Play):
+            if isinstance(inst[1].pulse, GaussianSquare) and isinstance(
+                inst[1].channel, DriveChannel
+            ):
+                return True
+
+        return False
+
+    def get_calibration(self, params: List, qubits: List) -> Schedule:
+        """
+        Builds the calibration schedule for the RZXGate(theta) without echos.
+        Args:
+            params: Parameters of the RZXGate(theta). I.e. params[0] is theta.
+            qubits: List of qubits for which to get the schedules. The first qubit is
+                the control and the second is the target.
+        Returns:
+            schedule: The calibration schedule for the RZXGate(theta).
+        Raises:
+            QiskitError: If the control and target qubits cannot be identified, or the backend
+                does not support a cx gate between the qubits, or the backend does not natively
+                support the specified direction of the cx.
+        """
+        theta = params[0]
+        q1, q2 = qubits[0], qubits[1]
+
+        if not self._inst_map.has("cx", qubits):
+            raise QiskitError(
+                "This transpilation pass requires the backend to support cx "
+                "between qubits %i and %i." % (q1, q2)
+            )
+
+        cx_sched = self._inst_map.get("cx", qubits=(q1, q2))
+        rzx_theta = Schedule(name="rzx(%.3f)" % theta)
+
+        if theta == 0.0:
+            return rzx_theta
+
+        control, target = None, None
+
+        for _, inst in cx_sched.instructions:
+            # Identify the compensation tones.
+            if isinstance(inst.channel, DriveChannel) and isinstance(inst, Play):
+                if isinstance(inst.pulse, GaussianSquare):
+                    target = inst.channel.index
+                    control = q1 if target == q2 else q2
+
+        if control is None:
+            raise QiskitError("Control qubit is None.")
+        if target is None:
+            raise QiskitError("Target qubit is None.")
+
+        # Get the filtered Schedule instructions for the CR gates and compensation tones.
+        crs = cx_sched.filter(*[self._filter_control]).instructions
+        rotaries = cx_sched.filter(*[self._filter_drive]).instructions
+
+        # Stretch/compress the CR gates and compensation tones.
+        cr = self.rescale_cr_inst(crs[0][1], 2 * theta)
+        rot = self.rescale_cr_inst(rotaries[0][1], 2 * theta)
+
+        # Build the schedule for the RZXGate without the echos.
+        rzx_theta = rzx_theta.insert(0, cr)
+        rzx_theta = rzx_theta.insert(0, rot)
+        rzx_theta = rzx_theta.insert(0, Delay(cr.duration, DriveChannel(control)))
+
+        # Reverse direction of the ZX.
+        if control == qubits[0]:
+            return rzx_theta
+        else:
+            raise QiskitError("Reverse direction not supported.")
```

Comment:
```suggestion
        return rzx_theta
```
looking at this it makes more sense to do the test earlier to fail faster in case of an error.